provider/github/watcher.go

```diff
@@ -146,23 +234,44 @@ func (w *Watcher) doEventRequest(ctx context.Context, username, repository strin
 		return resp, nil, err
 	}
 
+	interval := w.newInterval(resp)
+	// obey poll interval
 	secs, _ := strconv.Atoi(resp.Response.Header.Get("X-Poll-Interval"))
-	w.pollInterval = time.Duration(secs) * time.Second
+	pollLimit := time.Duration(secs) * time.Second
+	if pollLimit > interval {
+		interval = pollLimit
+	}
+	w.eventsPollInterval = interval
 
 	if isStatusNotModified(resp.Response) {
 		return nil, nil, NoErrNotModified.New()
 	}
 
-	log.With(log.Fields{
-		"remaining-requests": resp.Rate.Remaining,
-		"reset-at":           resp.Rate.Reset,
-		"poll-interval":      w.pollInterval,
-		"events":             len(events),
-	}).Debugf("Request to events endpoint done.")
+	w.responseLogger(resp).With(log.Fields{"poll-interval": w.eventsPollInterval}).
+		Debugf("Request to events endpoint done with %d events.", len(events))
 
 	return resp, events, err
 }
 
+func (w *Watcher) newInterval(resp *github.Response) time.Duration {
+	remaining := resp.Rate.Remaining / 2 // we call 2 endpoints for each repo
```

Comment:
What about requests made to post comments, will that decrease the remaining rate? Should we accommodate some margin here for requests that are to polling?

Reply:
Any other calls will decrease remaining. What margin do you want to add?

Reply:
:thinking:  I'm not sure how to solve it, but I'm a bit concerned with the last few calls before `Reset`. We are adjusting the polling without accounting for any other calls, so the last calls to post comments may fail hitting our rate limit.

Reply:
Ah. I see what you mean now!
Though I can't predict how many comments will be posted.
The poster must handle rate limit correctly. 

I don't think solving this problem is within the scope of this PR.
I also don't think we should fix it right now. I would better wait for the multi-token support first.
As a quick workaround we can always change `minInterval` from 1 second to 2,3,4,5 and it will give us extra calls.